### PR TITLE
1password 1pif: import password history

### PR DESCRIPTION
**Pull request by spiderpug**
Tested using 1pif export of 1Password 6

Export format does not include the `passwordHistory` key if the password was never changed, so checking for presence should be good enough IMO. 

solves part of #18

### CODE REVIEW

File: src/importers/onepassword1PifImporter.ts [hunk 1]
```diff
@@ -120,6 +121,15 @@ export class OnePassword1PifImporter extends BaseImporter implements Importer {
                     }
                 });
             }
+            if (item.secureContents.passwordHistory != null) {
+                cipher.passwordHistory = cipher.passwordHistory || [];
+                item.secureContents.passwordHistory.forEach((entry: any) => {
+                    const phv = new PasswordHistoryView();
+                    phv.password = entry.value;
+                    phv.lastUsedDate = new Date(entry.time * 1000);
```
Comment (spiderpug) [inline src/importers/onepassword1PifImporter.ts]: Is there utility code / a smarter way to create the correct timestamp here?

Comment (kspearrin) [reply src/importers/onepassword1PifImporter.ts]: No, this should be fine.

**Comment by spiderpug** [comment]
Is it fine to implement `passwordHistory` sorting / truncation in here?

https://github.com/bitwarden/jslib/blob/c17e8b458cb64e767559b24c9c69a7776bcb0ea4/src/importers/baseImporter.ts#L285

**Comment by kspearrin** [comment]
We probably want to have it somewhere central, like the `ImportService` class.

**Comment by kspearrin** [comment]
Here's what win opvault looks like for password history:

```
{
   "category":"001",
   "created":1544823719,
   "hmac":"NtyBmTTPOb88HV3JUKPx1xl/vcMhac9kvCfe/NtszY0=",
   "k":"XC/z20QveYCoV8xQ4tCJZZp/uum77xLkMSMEhlUULndryXgSmkG+VBtkW7AfuerfKc8Rtu43a4Sd078j7XfZTcwUCEKtBECUTDNbEgv4+4hoFVk1EzZgEUy/0bW1Ap+jNLmmdSU9h74+REu6pdxsvQ==",
   "tx":1553395669,
   "updated":1553395669,
   "uuid":"528AB076FB5F4FBF960884B8E01619AC",
   "overview":{
      "title":"Google",
      "URLs":[
         {
            "u":"google.com"
         }
      ],
      "url":"google.com",
      "ps":26,
      "ainfo":"googluser"
   },
   "details":{
      "passwordHistory":[
         {
            "value":"oldpass2",
            "time":1553394449
         },
         {
            "value":"oldpass1",
            "time":1553394457
         }
      ],
      "fields":[
         {
            "type":"T",
            "id":"username",
            "name":"username",
            "value":"googluser",
            "designation":"username"
         },
         {
            "type":"P",
            "id":"password",
            "name":"password",
            "value":"12345678901",
            "designation":"password"
         }
      ],
      "notesPlain":"This is a note\r\n\r\nline1\r\nline2",
      "sections":[
         {
            "title":"test",
            "name":"1214FD88CD30405D9EED14BEB4D61B60",
            "fields":[
               {
                  "k":"string",
                  "n":"6CC3BD77482D4559A4B8BB2D360F821B",
                  "v":"fgfg",
                  "t":"fgggf"
               },
               {
                  "k":"concealed",
                  "n":"5CFE7BCAA1DF4578BBF7EB508959BFF3",
                  "v":"dfgdfgfdg",
                  "t":"pwfield"
               }
            ]
         }
      ]
   }
}
```

**Comment by spiderpug** [comment]
> Here's what win opvault looks like for password history:

Thanks. I added this to the implementation.

> We probably want to have it somewhere central, like the ImportService class.

I have a hard time coming up with a way to make this fit in there. AFAICS this service class is preparing API requests mostly and it does not touch the data (yet). I can probably create `prepareCiphers(importResult.ciphers)` before calling `this.postImport(...)`, if you think that's okay.

I can also make `ImportService` call `baseImporter::cleanupChiper` for all ciphers if you think an importer implementation might miss this call.

What do you think?

**Comment by kspearrin** [comment]
I think that this is good for now. I'll think more about the best way to sanitize the data further before doing the import API.

**Comment by kspearrin** [comment]
@spiderpug FYI: https://github.com/bitwarden/jslib/commit/58c34b896c214cc599ecb106adce6aee8a400802

**Pull request merged**